### PR TITLE
pm: Introduce CONFIG_PM_DEVICE_RUNTIME_EXCLUSIVE

### PR DESCRIPTION
**Pull request by carlocaione**
Sometimes we want to entirely decouple the system PM from the device PM,
leaving the devices to manage its own power states using the runtime PM.

This is currently not possible because the suspend / resume code path is
triggering the device PM hooks even when the runtime PM is enabled.

Introduce a new PM_DEVICE_RUNTIME_EXCLUSIVE symbol to allow the platform
to skip the device PM triggers on suspend / resume.

Signed-off-by: Carlo Caione <ccaione@baylibre.com>

### CODE REVIEW

File: subsys/pm/Kconfig [hunk 1]
```diff
@@ -103,6 +103,13 @@ config PM_DEVICE_RUNTIME
 	  enabled, devices can be suspended or resumed based on the device
 	  usage even while the CPU or system is running.
 
+config PM_DEVICE_RUNTIME_EXCLUSIVE
+	depends on PM_DEVICE_RUNTIME
```
Comment (gmarull) [inline subsys/pm/Kconfig:107]: nit: it should probably be `default y`

Comment (carlocaione) [reply subsys/pm/Kconfig:107]: I think we should be cautious here and not setting this to `y` to not disrupt the current behavior.

**Comment by ceolin** [comment]
@carlocaione I think that is ok. Just to let you know, I'm planning to officially decouple device pm from system pm. The main reasons are:

1) The kernel constantly trigger system PM, sometimes for a really short period. Suspend / resume devices in this circumstances may result in more energy consumption than just let them up.
2) The time required to suspend / resume devices is not accounted by the kernel. Do it properly brings complexity that is hard to justify
3) Complex drivers sometimes do this asynchronously which big problem for system pm
4) Promotes device runtime PM :)

**Review by ceolin — DISMISSED** [review]

File: subsys/pm/Kconfig (hunk 1 above)
Comment (gmarull) [reply subsys/pm/Kconfig:107]: Note that device PM coupled with system PM has issues now (to the point it's not much usable in practice), so I think it makes sense defaulting it to `y` when users enable runtime PM

**Review by gmarull — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Review by gmarull — APPROVED** [review]
`approvals++;`

**Review by ceolin — APPROVED** [review]

**Pull request merged**